service/src/main/java/org/broadinstitute/listener/relay/inspectors/TokenChecker.java

```diff
@@ -0,0 +1,61 @@
+package org.broadinstitute.listener.relay.inspectors;
+
+import com.auth0.jwt.JWT;
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.time.Instant;
+import java.util.Optional;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class TokenChecker {
+  private final HttpClient httpClient = HttpClient.newHttpClient();
+  private final String GOOGLE_OAUTH_SERVER = "https://www.googleapis.com/oauth2/v1/tokeninfo?access_token=";
+  private final Logger logger = LoggerFactory.getLogger(SamResourceClient.class);
+
+  public OauthInfo getOauthInfo(String token) throws IOException, InterruptedException {
+    var jwtExpiration = tryDecodeAsB2cToken(token);
+
+    var now = Instant.now();
+
+    if (jwtExpiration.isPresent()) {
+      if(jwtExpiration.get().isAfter(now))
+        return new OauthInfo(Optional.of(jwtExpiration.get()), "");
```

Comment:
this is redundant, I think
```suggestion
        return new OauthInfo(jwtExpiration, "");
```